components/d2l-quick-eval/activities-list/d2l-quick-eval-activities-list.js

```
@@ -0,0 +1,73 @@
+import {html, PolymerElement} from '@polymer/polymer/polymer-element.js';
+import {QuickEvalLocalize} from '../QuickEvalLocalize.js';
+import '../activity-card/d2l-quick-eval-activity-card.js';
+
+class D2LQuickEvalActivitiesList extends QuickEvalLocalize(PolymerElement) {
+	static get is() { return 'd2l-quick-eval-activities-list'; }
+	static get template() {
+		return html`
+			<div>
+				<dom-repeat items="[[_courses]]" as="c">
+					<template>
+						<h3>[[c.name]]</h3>
+						<dom-repeat items="[[c.activities]]" as="a">
+							<template>
+								<d2l-quick-eval-activity-card
+								assigned="[[a.assigned]]"
+								completed="[[a.completed]]"
+								published="[[a.published]]"
+								evaluated="[[a.evaluated]]"
+								unread="[[a.unread]]"
+								resubmitted="[[a.resubmitted]]"
+								due-date="[[a.dueDate]]"
+								activity-type="[[localize(a.activityType)]]"
+								activity-name-href="[[a.activityNameHref]]"
+								token="[[token]]"></d2l-quick-eval-activity-card>
+							</template>
+						</dom-repeat>
+					</template>
+				</dom-repeat>
+			</div>
+		`;
+	}
+	static get properties() {
+		return {
+			activities: {
+				type: Array,
+				value: [
+					// {
+					// 	courseName: '',
+					// 	assigned: 0,
+					// 	completed: 0,
+					// 	published: 0,
+					// 	evaluated: 0,
+					// 	unread: 0,
+					// 	resubmitted: 0,
+					// 	dueDate: '',
+					// 	activityType: '',
+					// 	activityNameHref: ''
+					// }
+				]
+			},
+			_courses: {
+				type: Array,
+				computed: '_groupByCourse(activities)'
+			},
+			token: {
+				type: String,
+				value: ''
```

does this default value serve a purpose?

---

It's a habit to define an explicit "blank" value so that I don't run up against weird `undefined` and/or `null` behaviours down the line. Should it be removed?

---

I think so. Either we expect weird behaviour, and we should handle that expected behaviour, or we're not sure what will happen, and in that case I think we want to fail fast and loudly. Unless there's a specific case we're guarding against I think we should just remove the defaults.